### ISSUE TITLE
LTS and Contributor Guidelines

### ISSUE THREAD

**Issue by okwme** [cosmos/ethermint#719]
__Proposal:__ Add Contributor guidelines and protocol for v0.39.x LTS support

__Current behavior:__  Currently work is going towards upgrading to v0.40.x in order to include IBC but there are teams like OKEX who [are already contributing](https://github.com/cosmos/ethermint/issues/707) in supporting v0.39.x for at least 6 months. There are other teams in the ecosystem who are taking a conservative timeline to upgrading to stargate and provding a stable version of ethermint for them could be a good idea.

__Desired behavior:__ Creating a protocol so that multiple teams have a clear path for contributing to the shared code base with maybe some agreements around where responsibilities lie. The Cosmos SDK could be a model for this protocol as outlined in the [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md) and groups of developers to be part of assigning point release committees in a similar fashion.

__Use case:__ So that the quality of Ethermint improves with multiple coordinated efforts instead of all contributions coming from a single organization

Requests may be closed if we're not actively planning to work on them.

**Comment by okwme** [comment]
cc @zhongqiuwoo  @ColinSchwarz @fedekunze

**Comment by fedekunze** [comment]
The issue with v0.39 LTS is that is is not production ready and still suffers from performance degradation. There are also a significant amount of bugs and improvements that would need to be made to account for a v0.39 stable. For example the fix for #667 would require a large refactor of the evm module (see https://github.com/cosmos/ethermint/pull/677#issuecomment-752055507) which would necessarily introduce breaking changes.

There are also several issues that haven't been addressed yet like ledger compatibility, more e2e test for different opcodes and smart contracts, gas refund, gas estimation checks #497,  docs, etc etc. 

In summary, there are 2 main options for this:

- Stability on v0.39 by addressing the issues outlined above
- Stability on v0.40 (stargate) which will require significant updates to the existing PRs to migrate the repo to that version. Which I won't be available to help because my contract with ChainSafe ends

**Comment by hxrts** [comment]
cc @zhongqiuwood typo in your handle above^
